src/sys/time.rs

```diff
@@ -81,9 +81,24 @@ impl From<Duration> for TimeSpec {
     }
 }
 
-impl From<TimeSpec> for Duration {
-    fn from(timespec: TimeSpec) -> Self {
-        Duration::new(timespec.0.tv_sec as u64, timespec.0.tv_nsec as u32)
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct DurationFromTimeSpecError {}
+
+impl fmt::Display for DurationFromTimeSpecError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Conversion to Duration failed")
+    }
+}
+
+impl std::error::Error for DurationFromTimeSpecError {}
+
+
+impl TryFrom<TimeSpec> for Duration {
+    type Error = DurationFromTimeSpecError;
+
+    fn try_from(value: TimeSpec) -> Result<Self, Self::Error> {
+        let nanos = u64::try_from(value.num_nanoseconds()).map_err(|_| DurationFromTimeSpecError{})?;
```

Comment:
This conversion will fail for durations of 585 years or more.  While that's a lot, it's still an unnecessary limitation.